### DISCUSSION TITLE
Open hyperlink in new tab by default (website)

### DISCUSSION THREAD

**Discussion by pat-alt** [quarto-dev/quarto-cli#3169] [Q&A] [answered]
Sometimes it's useful to host documents or even entire Reveal.js presentations on a website. It's currently possible to use hyperlinks in order to give users access to such static content. For example, I have a link to my CV (rendered using `pagedown::html_resume` on my website. The problem is that as soon as users click on the link, this will take them to static document and the website's navigation bar will disappear. To avoid this, it would be nice to have the option to have these links open in a new tab by default. It's possible to do this using raw html (see [here](https://stackoverflow.com/a/4425223/6681707)), but where possible I try to rely on Markdown (and Quarto extensions). 

Is this something that would be easy enough to add and considered valuable?

Thanks!

**Comment by dragonstyle** [comment, marked as answer]
You can also use markdown links like so:

```markdown
[link](https://example.com){target="_blank"}
```

**Reply by cscheid** [reply]
And a filter like this could presumably work as well:

```lua
-- newpagelink.lua

function Link(link)
  link.attributes["target"] = "_blank"
  return link
end
```

**Reply by pat-alt** [reply]
Ahh nice, I had tried `[link](https://example.com){:target="_blank"}` with the colon added. That's neat, thanks!

@cscheid in this case, that will apply to all links I suppose?

**Reply by cscheid** [reply]
Yes, though you do have to remember to add

```yaml
filters:
  - newpagelink.lua
```

to either the document or project metadata.

**Reply by pat-alt** [reply]
Nice, thanks both!

**Reply by davidwyld** [reply]
I've turned this answer into an extension, here: https://github.com/davidwilby/openlinksinnewpage

**Reply by jgunstone** [reply]
is it possible to add additional HTML directives?
i.e. 
```
[`toolbox.maxfordham.com`](https://toolbox.maxfordham.com/){target="_blank",menubar="no",scrollbars="yes",resizable="yes",top="500",left="500",width="1200",height="600"toolbar="no"}
```
^tried this but it didn't work...

**Reply by mcanouil** [reply]
You are using bad syntax to set options and mixing things.

```
[`quarto.org`](https://quarto.org/){target="_blank" menubar="no" scrollbars="yes" resizable="yes" top="500" left="500" width="1200" height="600" toolbar="no"}
```

Would be the correct syntax, regardless of the HTML validity of your options.

It will produce:

```html
<p><a href="https://quarto.org/" target="_blank" data-menubar="no" data-scrollbars="yes" data-resizable="yes" data-top="500" data-left="500" width="1200" height="600" data-toolbar="no"><code>quarto.org</code></a></p>
```

**Reply by jgunstone** [reply]
thanks for this - the syntax you shared works. 
unfortunately it's not doing quite what I'm looking for (my original q was wrong)
I'm trying to reproduce this HTML: 
```html
<a href="#" onclick="window.open('https://www.google.com', '_blank', 'toolbar=no,menubar=no,scrollbars=yes,resizable=yes,top=500,left=500,width=1200,height=1000' )"><b>🔗google🔗</b></a>
```
so that it opens a new window without a URL address bar... 
do you know if this is possible ?

**Reply by mcanouil** [reply]
Use the syntax showed previously. `data-` won't be added if the option is a valid HTML option.
Or use raw code `` `whatever html code`{=html} ``.

**Reply by jgunstone** [reply]
great - thanks - this sorted me out

**Reply by aloboa** [reply]
Is it possible to force the link to open in a new window rather than in a new tab? I understand this might not be a common need, but in some cases, such as maps, it is better to get them opened in a different window.

**Reply by mcanouil** [reply]
Adapt the code shared with the right target that you can easily find online.
